components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/bean/SSOAgentConfig.java

```diff
@@ -190,10 +214,58 @@ public KeyStore getKeyStore() throws org.wso2.carbon.identity.sso.agent.exceptio
     }
 
     public void setKeyStore(KeyStore keyStore) {
+
         this.keyStore = keyStore;
     }
+
     public void initConfig(Properties properties) throws SSOAgentException {
 
+        char[] key = new char[]{};
+        String decodedPassword;
+        boolean isReadFile = false;
+        String pathToFile = SSOAgentConstants.SSOAgentConfig.SECRETKEY_FILEPATH;
+
+        for (Map.Entry<Object, Object> entry : properties.entrySet()) {
+            if (String.valueOf(entry.getValue()).startsWith("Enc:")) {
+                if (System.getProperty("key") == null) {
+                    if (!isReadFile) {
+                        File file = new File(pathToFile);
+                        if (file.exists()) {
+                            try (BufferedReader brTest = new BufferedReader(new FileReader(file))) {
+                                key = brTest.readLine().toCharArray();
+                                isReadFile = true;
+                                file.delete();
+                            } catch (IOException e) {
+                                LOGGER.log(Level.SEVERE, "Error while reading the file "+ pathToFile);
+                            }
+                        } else {
+                            LOGGER.log(Level.SEVERE, "Can't find the file for retrieving the key");
```

Comment:
Make the log context aware. What is the 'key' and 'file' here? In which scenarios this error can occur?